From 86313bab12214247a7bfd37aae8ee993e787c33a Mon Sep 17 00:00:00 2001
From: Gustavo Romero <gustavo.romero@linaro.org>
Date: Thu, 14 Mar 2024 12:37:09 +0000
Subject: [PATCH 05/16] cmake: emu: Allow QEMU to serve serial on a TCP/IP port

This commit enables users to pass -DQEMU_SOCKET=y for 'west' when
selecting to run the build on a QEMU VM. This allows the target's serial
port to be served to the host via a TCP/IP port for convenience, easing
communication with the target from the external world.

Signed-off-by: Erik Schilling <erik.schilling@linaro.org>
Signed-off-by: Gustavo Romero <gustavo.romero@linaro.org>
---
 cmake/emu/qemu.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmake/emu/qemu.cmake b/cmake/emu/qemu.cmake
index 25acaf02fdc..72e1196165c 100644
--- a/cmake/emu/qemu.cmake
+++ b/cmake/emu/qemu.cmake
@@ -67,6 +67,9 @@ elseif(QEMU_PIPE)
   foreach(target ${qemu_targets})
     list(APPEND PRE_QEMU_COMMANDS_FOR_${target} COMMAND ${CMAKE_COMMAND} -E touch ${QEMU_PIPE})
   endforeach()
+elseif(QEMU_SOCKET)
+  # Serve serial console on a TCP/IP port.
+  list(APPEND QEMU_FLAGS -chardev socket,id=con,mux=on,server=on,host=127.0.0.1,port=4321)
 else()
   # Redirect console to stdio, used for manual debugging.
   list(APPEND QEMU_FLAGS -chardev stdio,id=con,mux=on)
-- 
2.43.0

